Source/Core/PolygonGeometry.js

```diff
@@ -661,30 +673,48 @@ define([
         var results = PolygonGeometryLibrary.polygonsFromHierarchy(polygonHierarchy);
         var hierarchy = results.hierarchy;
         var polygons = results.polygons;
+        var i;
 
         if (polygons.length === 0) {
             return undefined;
         }
 
-        outerPositions = polygons[0];
+        outerPositions = polygons[0].slice();
```

Comment:
Each of the places we call PolygonGeometry should actually already know this information and could pass it in.